modules/cluster-logging-collector-external.adoc

```diff
@@ -14,152 +14,129 @@ ifdef::openshift-origin[]
-#     retry_forever true
-#     # the systemd journald 0.0.8 input plugin will just throw away records if the buffer
-#     # queue limit is hit - 'block' will halt further reads and keep retrying to flush the
-#     # buffer to the remote - default is 'exception' because in_tail handles that case
-#     overflow_action "#{ENV['BUFFER_QUEUE_FULL_ACTION'] || 'exception'}"
-#   </buffer>
-#   <server>
-#     host server.fqdn.example.com  # or IP
-#     port 24284
-#   </server>
-#   <server>
-#     host 203.0.113.8 # ip address to connect
-#     name server.fqdn.example.com # The name of the server. Used for logging and certificate verification in TLS transport (when host is address).
-#   </server>
-# </store>
+<store>
+  @type forward
+  <security>
+    self_hostname ${hostname} # ${hostname} is a placeholder.
+    shared_key "fluent-receiver"
+  </security>
+  transport tls
+  tls_verify_hostname false           # Set false to ignore server cert hostname.
+
+  tls_cert_path '/etc/ocp-forward/ca-bundle.crt'
+  <buffer>
+    @type file
+    path '/var/lib/fluentd/secureforwardlegacy'
+    queued_chunks_limit_size "#{ENV['BUFFER_QUEUE_LIMIT'] || '1024' }"
+    chunk_limit_size "#{ENV['BUFFER_SIZE_LIMIT'] || '1m' }"
+    flush_interval "#{ENV['FORWARD_FLUSH_INTERVAL'] || '5s'}"
+    flush_at_shutdown "#{ENV['FLUSH_AT_SHUTDOWN'] || 'false'}"
+    flush_thread_count "#{ENV['FLUSH_THREAD_COUNT'] || 2}"
+    retry_max_interval "#{ENV['FORWARD_RETRY_WAIT'] || '300'}"
+    retry_forever true
+    # the systemd journald 0.0.8 input plugin will just throw away records if the buffer
+    # queue limit is hit - 'block' will halt further reads and keep retrying to flush the
+    # buffer to the remote - default is 'exception' because in_tail handles that case
+    overflow_action "#{ENV['BUFFER_QUEUE_FULL_ACTION'] || 'exception'}"
+  </buffer>
+  <server>
+    host fluent-receiver.openshift-logging.svc  # or IP
+    port 24224
+  </server>
+</store> 
 ----
 
 .Procedure
 
 To send a copy of Fluentd logs to an external log aggregator:
 
-. Edit the `secure-forward.conf` section of the Fluentd configuration map:
+. Create a configuration map called `secure-forward` in the `openshift-logging` namespace:
 +
-----
-$ oc edit configmap/fluentd -n openshift-logging
-----
-
-. Enter the name, host, and port for your external Fluentd server:
+.. Configure the secrets and TLS information:
 +
 [source,yaml]
 ----
-#   <server>
-#     host server.fqdn.example.com  # or IP
-#     port 24284
-#   </server>
-#   <server>
-#     host 203.0.113.8 # ip address to connect
-#     name server.fqdn.example.com # The name of the server. Used for logging and certificate verification in TLS transport (when host is address).
-#   </server>
+ <store>
+  @type secure_forward
+
+  self_hostname ${hostname} <1>
+  shared_key <SECRET_STRING> <2>
+
+  transport tls <3>
+
+  tls_verify_hostname true <4>
+  tls_cert_path /path/for/certificate/ca_cert.pem <5>
```

Comment:
Seems the tls_cert_path is hard coded to `/etc/ocp-forward` https://github.com/openshift/cluster-logging-operator/blob/master/pkg/k8shandler/fluentd.go#L287  @jcantrill  could you please help to confirm this ?

Reply:
The directory is 'hardcoded' but the actual cert is not. In this example it may be better to change to `/etc/ocp-forward/ca_cert.pem`